Configure VS Code Python settings and recommended extensions

jtbandes opened the pull request:
cc @cclauss

Seems our CI has converged on bandit and flake8.

jtbandes committed:
Configure VS Code Python settings and recommended extensions

amacneil commented on .gitignore:
```diff
@@ -5,7 +5,12 @@
+
+.vscode/*
+!.vscode/extensions.json
+!.vscode/settings.json
```
Just remove from gitignore entirely? What else would live in this directory?

jtbandes replied:
I have a c_cpp_properties.json in there…must’ve been automatically created? I could delete it I suppose, but it’s annoying to have it constantly show as an untracked file.

amacneil commented:
Suggest waiting until we run black on the whole repo and then enable formatOnSave

cclauss commented:
I would like to recommend that we use https://pre-commit.com to run our fastest lint jobs locally.

Additionally, I would like to recommend that we use https://pre-commit.ci to run those lint jobs instead of GitHub Actions.  The advantage of pre-commit.ci is that it will auto-format a pull request.

For example, see the workflow at https://github.com/kornia/kornia/pull/1210#commits-pushed-3e05ead where I make my commits and then pre-commit.ci comes in after me and automatically blackens my modifications so I don't have to.

jtbandes committed:
Merge remote-tracking branch 'origin/ros2' into jacob/vscode-config

jtbandes committed:
Move some pre-commit configuration into pyproject.toml; enable format on save

jtbandes committed:
trailing newline

jtbandes commented:
@cclauss Now that we are using pre-commit in this repo, I updated this PR to move some of the pre-commit tool configs into pyproject.toml so they can also be read when running the tools directly.

cclauss commented on .vscode/settings.json:
```diff
@@ -4,4 +4,7 @@
+    "[python]": {
+        "editor.formatOnSave": true
+    },
+}
```
Please add one carriage return (\n).

jtbandes committed:
Keep bandit args in .pre-commit-config.yaml as well as .vscode/settings.json

Bandit will soon support configuration via pyproject.toml but a version supporting this has not yet been released. See https://github.com/PyCQA/bandit/issues/550
